subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt_state.c

```diff
@@ -422,6 +506,51 @@ img_mgmt_state_read(struct smp_streamer *ctxt)
 	return ok ? MGMT_ERR_EOK : MGMT_ERR_EMSGSIZE;
 }
 
+static int img_mgmt_set_next_boot_slot_common(int slot, int active_slot, bool confirm)
+{
+	const struct flash_area *fa;
+	int area_id = img_mgmt_flash_area_id(slot);
+	int rc = 0;
+
+	if (flash_area_open(area_id, &fa) != 0) {
+		return IMG_MGMT_ERR_FLASH_OPEN_FAILED;
+	}
+
+	rc = boot_set_next(fa, slot == active_slot, confirm);
+	if (rc != 0) {
+		/* Failed to set next slot for boot as desired */
+		LOG_ERR("Faled boot_set_next with code %d, for slot %d,"
+			" with active slot %d and confirm %d",
+			 rc, slot, active_slot, confirm);
+
+		/* Translate from boot util error code to IMG mgmt group error code */
+		if (rc == BOOT_EFLASH) {
+			rc = IMG_MGMT_ERR_FLASH_WRITE_FAILED;
+		} else if (rc == BOOT_EBADVECT) {
+			rc = IMG_MGMT_ERR_INVALID_IMAGE_VECTOR_TABLE;
+		} else if (rc == BOOT_EBADIMAGE) {
+			rc = IMG_MGMT_ERR_INVALID_IMAGE_HEADER_MAGIC;
+		} else {
+			rc = IMG_MGMT_ERR_UNKNOWN;
+		}
+	}
+	flash_area_close(fa);
+
+#if defined(CONFIG_MCUMGR_GRP_IMG_STATUS_HOOKS)
+	if (slot == active_slot && confirm) {
+		int32_t err_rc;
+		uint16_t err_group;
+
+		/* Confirm event is only sent for active slot */
+		(void)mgmt_callback_notify(MGMT_EVT_OP_IMG_MGMT_DFU_CONFIRMED, NULL, 0, &err_rc,
```

Comment:
Not directly related to the PR, but should be notify also if `boot_set_next` fails ?

Reply:
Thanks for fixing it